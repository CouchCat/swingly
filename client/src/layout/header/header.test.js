import React from 'react';
import { render, fireEvent } from "../../test-utils";
import ConnectedHeader, { Header } from './header.component.jsx';
import { MemoryRouter } from "react-router-dom";
import { createMemoryHistory } from 'history';

describe('Header', () => {
  let currentUserMock = null;
  let initialEntries = "/";
  let mockProps = null;

  const applyMockProps = () => {
    mockProps = {
      currentUser: currentUserMock,
      history: createMemoryHistory({ initialEntries: [initialEntries] })
    }
  }

  afterEach(() => {
    currentUserMock = null;
    initialEntries = "/";
    jest.clearAllMocks();
  });

  it('should render the Header', () => {
    applyMockProps();
    const renderResult = render(<Header {...mockProps} />)

    expect(renderResult).toMatchSnapshot();
  });

  describe("Header left button", () => {
    it("should display the navigation Button instead", () => {
      applyMockProps();
      const { container } = render(<ConnectedHeader {...mockProps} />);
      expect(container.querySelector("[aria-label='navigation button']")).toBeTruthy();
    });

    it("should display a 'Back' Button instead of navigation button", () => {
      initialEntries = "/profile";
      applyMockProps();
      const { queryByText } = render(
        <MemoryRouter initialEntries={[initialEntries]}>
          <ConnectedHeader {...mockProps} />
        </MemoryRouter>
      );
      
      expect(queryByText("Back")).toBeTruthy();
    });

    it("should not display a 'Back' Button if user matches url query", () => {
      currentUserMock = "test_user";
      initialEntries = "/profile?user_id=test_user";
      applyMockProps();
      const { queryByText } = render(
        <MemoryRouter initialEntries={[initialEntries]}>
          <Header {...mockProps} />
        </MemoryRouter>
      );
      
      expect(queryByText("Back")).toBeFalsy();
    });

    it("should display a 'Events' Button instead of navigation button", () => {
      initialEntries = "/event-edit";
      applyMockProps();
      const { getByText } = render(
        <MemoryRouter initialEntries={[initialEntries]}>
          <ConnectedHeader {...mockProps} />
        </MemoryRouter>
      );
      
      expect(getByText("Events")).toBeTruthy();
    });

    it("should set navigation state to open on navigation button click", () => {
      applyMockProps();
      const { container } = render(<ConnectedHeader {...mockProps} />);
      const navigationButton = container.querySelector("[aria-label='navigation button']");

      fireEvent.click(navigationButton);

      expect(navigationButton.getAttribute("aria-expanded")).toBe("true");
    });
  });

  describe("Header right button", () => {
    it('should not render the ProfileImage component if user has not signed in.', () => {
      applyMockProps();
      const { queryByAltText } = render(<ConnectedHeader {...mockProps} />);
  
      expect(queryByAltText("profile image")).toBeNull();
    });

    it('should render the ProfileImage component if user has signed in.', () => {
      currentUserMock = "test_user";
      applyMockProps();
      const { getByAltText } = render(<Header {...mockProps} />);
  
      expect(getByAltText("profile image")).toBeTruthy();
    });

    it("should display the Sign in component on click if user has not signed in", () => {
      applyMockProps();
      const { container, queryByTestId } = render(<Header {...mockProps} />);
      const profileButton = container.querySelector("[aria-label='profile button']");

      fireEvent.click(profileButton);

      expect(queryByTestId("sign-in")).toBeTruthy();
    });

    it("should not display the Sign in component on click if user is signed in", () => {
      currentUserMock = "test_user";
      applyMockProps();
      const { container, queryByTestId } = render(<Header {...mockProps} />);
      const profileButton = container.querySelector("[aria-label='profile button']");

      fireEvent.click(profileButton);

      expect(queryByTestId("sign-in")).toBeFalsy();
    });

    it("should display the register component", () => {
      applyMockProps();
      const { container, queryByTestId, getByText } = render(<Header {...mockProps} />);
      const profileButton = container.querySelector("[aria-label='profile button']"); 

      fireEvent.click(profileButton);
      fireEvent.click(getByText("Register"));

      expect(queryByTestId("sign-up")).toBeTruthy();
    });

    it("should sign-up components on backdrop click", () => {
      applyMockProps();
      const { container, queryByTestId, getByTestId } = render(<Header {...mockProps} />);
      const profileButton = container.querySelector("[aria-label='profile button']"); 

      fireEvent.click(profileButton);
      fireEvent.click(getByTestId("backdrop"));

      expect(queryByTestId("sign-up")).toBeFalsy();
    });
  });
});