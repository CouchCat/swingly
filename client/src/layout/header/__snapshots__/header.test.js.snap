// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bwzfXH lcRQga"
      >
        <button
          aria-expanded="false"
          aria-label="navigation button"
          class="sc-bdVaJa jsLivE sc-htpNat xzQoi"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 fa-null fa-rotate-null fa-pull-null "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </button>
        <a
          class="sc-ifAKCX jurYiL"
          href="/"
        >
          Swingly 
        </a>
        <button
          aria-expanded="false"
          aria-label="profile button"
          class="sc-bdVaJa jsLivE sc-htpNat xzQoi"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user-circle fa-w-16 fa-2x fa-rotate-null fa-pull-null "
            data-icon="user-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              fill="currentColor"
            />
          </svg>
        </button>
      </header>
      <nav
        class="sc-EHOje guMuZC"
        data-testid="sidebar-container"
      >
        <div
          class="sc-bZQynM dBNBrd"
        >
          Navigation
        </div>
        <a
          class="sc-gzVnrw bbncCi"
          href="/"
          isactive="true"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-week fa-w-14 fa-null fa-rotate-null fa-pull-null "
            data-icon="calendar-week"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h288c8.8 0 16 7.2 16 16v64c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-64zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
          All Events
        </a>
        <a
          class="sc-gzVnrw fQdvQJ"
          data-testid="create-event"
          href="/"
          isactive="false"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-plus fa-w-14 fa-null fa-rotate-null fa-pull-null "
            data-icon="calendar-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M436 160H12c-6.6 0-12-5.4-12-12v-36c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48v36c0 6.6-5.4 12-12 12zM12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm316 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z"
              fill="currentColor"
            />
          </svg>
          Create Event
        </a>
        <a
          class="sc-gzVnrw fQdvQJ"
          href="/"
          isactive="false"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-users fa-w-20 fa-null fa-rotate-null fa-pull-null "
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
          Users
        </a>
        <a
          class="sc-gzVnrw fQdvQJ"
          href="/about"
          isactive="false"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-info-circle fa-w-16 fa-null fa-rotate-null fa-pull-null "
            data-icon="info-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              fill="currentColor"
            />
          </svg>
          About
        </a>
        <a
          class="sc-gzVnrw fQdvQJ"
          href="/error"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-null fa-rotate-null fa-pull-null "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
          Error Page
        </a>
        <button
          class="sc-bdVaJa gnRyba sc-gzVnrw fQdvQJ"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sun fa-w-16 fa-null fa-rotate-null fa-pull-null "
            data-icon="sun"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
              fill="currentColor"
            />
          </svg>
          Light Mode
        </button>
      </nav>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bwzfXH lcRQga"
    >
      <button
        aria-expanded="false"
        aria-label="navigation button"
        class="sc-bdVaJa jsLivE sc-htpNat xzQoi"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-w-14 fa-null fa-rotate-null fa-pull-null "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
      </button>
      <a
        class="sc-ifAKCX jurYiL"
        href="/"
      >
        Swingly 
      </a>
      <button
        aria-expanded="false"
        aria-label="profile button"
        class="sc-bdVaJa jsLivE sc-htpNat xzQoi"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-circle fa-w-16 fa-2x fa-rotate-null fa-pull-null "
          data-icon="user-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            fill="currentColor"
          />
        </svg>
      </button>
    </header>
    <nav
      class="sc-EHOje guMuZC"
      data-testid="sidebar-container"
    >
      <div
        class="sc-bZQynM dBNBrd"
      >
        Navigation
      </div>
      <a
        class="sc-gzVnrw bbncCi"
        href="/"
        isactive="true"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-calendar-week fa-w-14 fa-null fa-rotate-null fa-pull-null "
          data-icon="calendar-week"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h288c8.8 0 16 7.2 16 16v64c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-64zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
            fill="currentColor"
          />
        </svg>
        All Events
      </a>
      <a
        class="sc-gzVnrw fQdvQJ"
        data-testid="create-event"
        href="/"
        isactive="false"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-calendar-plus fa-w-14 fa-null fa-rotate-null fa-pull-null "
          data-icon="calendar-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M436 160H12c-6.6 0-12-5.4-12-12v-36c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48v36c0 6.6-5.4 12-12 12zM12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm316 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z"
            fill="currentColor"
          />
        </svg>
        Create Event
      </a>
      <a
        class="sc-gzVnrw fQdvQJ"
        href="/"
        isactive="false"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-users fa-w-20 fa-null fa-rotate-null fa-pull-null "
          data-icon="users"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
          />
        </svg>
        Users
      </a>
      <a
        class="sc-gzVnrw fQdvQJ"
        href="/about"
        isactive="false"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-info-circle fa-w-16 fa-null fa-rotate-null fa-pull-null "
          data-icon="info-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
          />
        </svg>
        About
      </a>
      <a
        class="sc-gzVnrw fQdvQJ"
        href="/error"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-null fa-rotate-null fa-pull-null "
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
        Error Page
      </a>
      <button
        class="sc-bdVaJa gnRyba sc-gzVnrw fQdvQJ"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sun fa-w-16 fa-null fa-rotate-null fa-pull-null "
          data-icon="sun"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
            fill="currentColor"
          />
        </svg>
        Light Mode
      </button>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
